gtests/net/mptcp/syscalls/client_selfconnect.pkt

```diff
@@ -0,0 +1,19 @@
+// connect() function, a client tries to connect to itself
```

Comment:
It is not about connecting to itself, but about simultaneous initiation. Please see the TCP RFC for more details: https://www.ietf.org/rfc/rfc9293.html#section-3.5-7

In the test, it is another peer that is injecting the SYN. Don't hesitate to take a packet capture when running the tests.

Can you then update this comment, and the commit message to mention that it is about "simultaneous connect", and not "connecting to itself"?